channeldb/graph.go

```diff
@@ -190,6 +190,36 @@ func (c *ChannelGraph) Database() *DB {
 	return c.db
 }
 
+// NumNodes returns the number of nodes currently in the graph
+func (c *ChannelGraph) NumNodes(tx *bbolt.Tx) (int, error) {
+
+	var result int
+	var err error
+
+	count := func(tx *bbolt.Tx) error {
+		nodes := tx.Bucket(nodeBucket)
+		if nodes == nil {
+			return ErrGraphNotFound
+		}
+
+		return nodes.ForEach(func(pubKey, _ []byte) error {
+			if len(pubKey) == 33 {
+				result += 1
+			}
```

Comment:
Is there also something with a different length in this bucket?

Reply:
`result++`

Reply:
Strange. Is this an old db where there are left-overs from something?

Reply:
I think there shouldn't be? The DB was generated using the path finding test harness code a few months ago.

Reply:
Can you find out what causes those non-33 byte keys to be written? I'd be hesitant to add the extra condition without fully understanding why.

Reply:
The only point I found is `SetSourceNode` uses a hardcoded `"source"` as key in this bucket. I can't tell if there's any other place.